circuit-lib/circuitlib-rs/src/gnark/helpers.rs

```diff
@@ -11,46 +11,65 @@ use serde_json::json;
 
 use crate::gnark::constants::{HEALTH_CHECK, SERVER_ADDRESS};
 
-pub fn spawn_gnark_server(path: &str, wait_time: u64) -> Child {
-    let server_process = Command::new("sh")
-        .arg("-c")
-        .arg(path)
-        .spawn()
-        .expect("Failed to start server process");
-
-    // Wait for the server to launch before proceeding.
-    thread::sleep(Duration::from_secs(wait_time));
+static IS_LOADING: AtomicBool = AtomicBool::new(false);
 
-    server_process
+pub async fn spawn_gnark_server(
+    path: &str,
+    restart: bool,
+    inclusion: bool,
+    non_inclusion: bool,
+    combined: bool,
```

Comment:
Are these settings (`inclusion`, `non_inclusion`, `combined`) mutually exclusive? It seems like they are, given that we are using `else if` few lines later, so even if many of them are enabled, we end up just with one of them being included in `arg`.

If your intention is to have them mutually exclusive, maybe it would make sense to have an enum?

```rust
pub enum ProofType {
    Inclusion,
    NonInclusion,
    Combined,
}
```

I think that would simplify the code. We would have just one argument instead of 3 and also `match` on that enum later.